pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -11,24 +11,33 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Optional
 
 
-class ClusterEnvironment:
+class ClusterEnvironment(ABC):
+    """ Specification of a cluster environment. """
 
-    def __init__(self):
-        self._world_size = None
+    @abstractmethod
+    def spawns_children(self) -> bool:
+        """ Whether the environment spawns the subprocesses or not. """
```

Comment:
Does this control whether at all children will be spawned or whether this is done by the environment or the plugin?

Nit: Maybe `spawn` is not the best term here, since we don't always spawn processes, but sometimes also fork them. Maybe `create` would be better?

Reply:
> Does this control whether at all children will be spawned or whether this is done by the environment or the plugin?

The latter: whether this is done by the environment or the plugin.
If `spawns_children` returns False, the DDPPlugin will lauch the subprocesses for each gpu. Otherwise we assume the process we are in is already a worker process.

I took your suggestion and renamed to `creates_children`, that's better.